libkatana_python_native/include/katana/python/TemplateSupport.h

```diff
@@ -0,0 +1,64 @@
+#ifndef KATANA_LIBKATANAPYTHONNATIVE_KATANA_PYTHON_TEMPLATESUPPORT_H_
+#define KATANA_LIBKATANAPYTHONNATIVE_KATANA_PYTHON_TEMPLATESUPPORT_H_
+
+#include <katana/python/TypeTraits.h>
+#include <pybind11/pybind11.h>
+
+namespace katana {
+
+// TODO(amp): Use template template parameters once we can rely on having them.
+
+/// Invoke `f<T>(m, name)` and assign the resulting class into `types[T]`.
+///
+/// \param m The module which should contain the resulting class.
+/// \param basename The base name of the class which will be suffixed with a type name.
+/// \param f The functor which instantiates the class.
+/// \param types A dict to fill with instantiations.
+/// \return The resulting class.
+template <typename T, typename F>
+pybind11::object
+InstantiateForType(
+    pybind11::module_ m, const std::string& basename, F f,
+    pybind11::dict types) {
+  pybind11::object dtype = katana::PythonTypeTraits<T>::default_dtype();
+  pybind11::object cls = f.template operator()<T>(
+      m, (basename + "[" + katana::PythonTypeTraits<T>::name + "]").c_str());
```

Comment:
I see. We are creating a different class for each template instantiation, is that correct? Kind of a "duh" moment since that's the idea behind templates but I see that we have to do this to bind things in Python.

Reply:
Yep. There are no generics in Python (there aren't generics in C++ either, but that's a different rant), so we need to create a new class for each instantiation.